rpc/core/net.go

```diff
@@ -13,19 +13,35 @@ import (
 // NetInfo returns network info.
 // More: https://docs.tendermint.com/master/rpc/#/Info/net_info
 func (env *Environment) NetInfo(ctx *rpctypes.Context) (*ctypes.ResultNetInfo, error) {
-	peersList := env.P2PPeers.Peers().List()
-	peers := make([]ctypes.Peer, 0, len(peersList))
-	for _, peer := range peersList {
-		peers = append(peers, ctypes.Peer{
-			NodeInfo:         peer.NodeInfo(),
-			IsOutbound:       peer.IsOutbound(),
-			ConnectionStatus: peer.Status(),
-			RemoteIP:         peer.RemoteIP().String(),
-		})
-	}
-	// TODO: Should we include PersistentPeers and Seeds in here?
-	// PRO: useful info
-	// CON: privacy
+	var peers []ctypes.Peer
+
+	switch {
+	case env.P2PPeers != nil:
+		peersList := env.P2PPeers.Peers().List()
+		peers = make([]ctypes.Peer, 0, len(peersList))
+		for _, peer := range peersList {
+			peers = append(peers, ctypes.Peer{
+				ID:  peer.ID(),
+				URL: peer.SocketAddr().String(),
+			})
+		}
+	case env.PeerManager != nil:
+		peerList := env.PeerManager.Peers()
```

Comment:
Looks like these methods take the `PeerManager`'s [mutex](https://github.com/tendermint/tendermint/blob/18784f7fe0fa316994dd94cbd84573d87d7958d4/internal/p2p/peermanager.go#L967) which means interacting with the RPC can lock the peer store. Do you think it's worth switching to a read-write lock in the `peermanager` to prevent it's methods from becoming lock starved?

Reply:
I'm not particularly worried about this, the legacy address book construct didn't use a RW lock, and the operation that's protected by the lock is something that's pretty quick and in memory.